### DISCUSSION TITLE
Crashlytics: Why doesn't Crashlytics accept an app parameter like other components?

### DISCUSSION THREAD

**Discussion by sergiocampama** [firebase/firebase-ios-sdk#8828] [General]
This makes it harder to use Crashlytics in apps that use multiple environments (like development, staging, production), since it always uses the default app. 

The other components have APIs like `Auth.auth(app: firebaseApp)` that can be used to inject different parameters.

**Comment by samedson** [comment]
Hey @sergiocampama, we envisioned customers would use the [configureWithOptions](https://firebase.google.com/docs/reference/ios/firebasecore/api/reference/Classes/FIRApp#+configurewithoptions:) API to support multiple environments, so they would set the options at startup for their dev / staging / etc apps.

We haven't invested in supporting multiple apps because the configureWithOptions flow has worked for users for this use case. For example, we haven't had requests for apps to dynamically change to their development environment at runtime. In addition, supporting multiple apps might be technically challenging with Crashlytics as we don't have an API to tell where to send crash reports.

Is there a use case we are missing?
